docs/localization/LOCALIZATION.md

```diff
@@ -74,343 +78,443 @@ export function localizationConfigurationFactory() {
+  locWithArg: string;
+}
+
+export const translations: SimpleHeaderPresTranslation = {
+  airline: 'o3r-simple-header-pres.airline',
+  motto: 'o3r-simple-header-pres.motto',
+  locWithArg: 'o3r-simple-header-pres.locWithArg'
+};
+```
 
-Use Angular built-in [DatePipe](https://angular.io/api/common/DatePipe), [DecimalPipe](https://angular.io/api/common/DecimalPipe) and [CurrencyPipe](https://angular.io/api/common/CurrencyPipe) and pass it current locale as the last parameter. The locale is read from **this.localizationService.getCurrentLanguage()**. To be able to use translateService, your component container should take benefit of dependency injection to get LocalizationService as parameter of constructor as well.
+### Component file
 
-For example if you want to localize simpleHeader component you will start by injecting TranslateService to the constructor of simple-header-pres.component.ts
-Your component also needs to implement Translatable interface which forces you to declare translations property. This property requires 3 decorators (`@Input() and @Localization(url)`. This will let you override localization keys from template and give some default localization to your component if you don't have your own to start with.
+Now to localize your component, you will start by injecting `LocalizationService` to the constructor.
+Your component also needs to implement the `Translatable` interface which forces you to declare the `translations` property. This property requires 2 decorators (`@Input()` and `@Localization(url)`, with `url` corresponding to the relative path of the localization file).
+This will let you override localization keys from the template and give your component some default localization if you don't have your own to start with.
 
 ```typescript
-import {Component, Input} from  '@angular/core';
-import {Localization, LocalizationService, Translatable} from  '@o3r/localization';
-import {SimpleHeaderPresTranslation, translations} from  './simple-header-pres.translation';
+import { Component, Input } from '@angular/core';
+import { Localization, LocalizationService, Translatable } from '@o3r/localization';
+import { SimpleHeaderPresTranslation, translations } from './simple-header-pres.translation';
 
 @Component({
   selector: 'o3r-simple-header-pres',
   styleUrls: ['./simple-header-pres.style.scss'],
   templateUrl: './simple-header-pres.template.html'
 })
 
-export  class SimpleHeaderPresComponent implements Translatable<SimpleHeaderPresTranslation>, ... {
+export class SimpleHeaderPresComponent implements Translatable<SimpleHeaderPresTranslation>, ... {
   /**
-  * Localization of the component
-  */
+   * Localization of the component
+   */
   @Input()
   @Localization('./simple-header-pres.localization.json')
   public translations: SimpleHeaderPresTranslation;
 
-
-  constructor(
-    public localizationService: LocalizationService) {
+  constructor(public localizationService: LocalizationService) {
```

Comment:
Not needed

Reply:
But the component need to have the import to `LocalizationModule`
else it will not have the access to the o3rTranslate pipe (it should be done by the schematic)